securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,69 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    data = request.get_json(force=True, silent=True)
+
+    if data is None:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
+
+    # Check if fingerprint is present in json request body
+    if 'fingerprint' not in data:
+        return jsonify({'success': False, 'error': 'fingerprint not in request', 'data': None}), 400
+
+    # Check if fingerprint value is string variable before query
+    if(isinstance(data['fingerprint'], str) == False):
+        return jsonify({'success': False, 'error': 'fingerprint value not valid string', 'data': None}), 400
+
+    # search for fingerprint in mailbox table, assign to variable
+    mailbox_num = Mailbox.query.filter_by(fingerprint=data['fingerprint']).first()
```

Comment:
We should have DB interactions in try except blocks so that if an exception is thrown we can return either the error that we are given or one of our choice.

Reply:
I attempted to put them in try blocks, but I need feedback to know if I did them right